academy/4-my-own-chain/stored-game.md

```diff
@@ -241,31 +241,31 @@ service Query {
+* The [query parameters](https://github.com/cosmos/b9-checkers-academy-draft/blob/9c9b90e/x/checkers/types/query.pb.go#L33-L35), as well as [how to serialize](https://github.com/cosmos/b9-checkers-academy-draft/blob/9c9b90e/x/checkers/types/query.pb.go#L501) and make them conform to the right Protobuf [`RequestQuery`](https://github.com/tendermint/tendermint/blob/1c34d17/abci/types/types.pb.go#L636-L641) interface.
 * The primary implementation of the gRPC service.
-* The implementation of all the storage [setters and getters](https://github.com/cosmos/b9-checkers-academy-draft/blob/d2a72b4ca9064a7e3e5014ba204ed01a4fe81468/x/checkers/keeper/grpc_query_stored_game.go) as extra functions in the keeper.
-* The implementation of the storage getters in the keeper [as they come from the gRPC server](https://github.com/cosmos/b9-checkers-academy-draft/blob/d2a72b4ca9064a7e3e5014ba204ed01a4fe81468/x/checkers/keeper/grpc_query_stored_game.go).
+* The implementation of all the storage [setters and getters](https://github.com/cosmos/b9-checkers-academy-draft/blob/9c9b90e/x/checkers/keeper/grpc_query_stored_game.go) as extra functions in the keeper.
+* The implementation of the storage getters in the keeper [as they come from the gRPC server](https://github.com/cosmos/b9-checkers-academy-draft/blob/9c9b90e/x/checkers/keeper/grpc_query_stored_game.go).
 
 ## Helper functions
 
-Your stored game stores are only strings. But you know that they represent `sdk.AccAddress` or even a game from the `rules` file. You are going to do operations on them. So how about adding helper functions to `StoredGame`?
+Your stored game stores are only strings, but they represent `sdk.AccAddress` or even a game from the `rules` file. Therefor, you add helper functions to `StoredGame` to do operations on them.
 
 1. Get the game `Creator`:
 
-    ```go [https://github.com/cosmos/b9-checkers-academy-draft/blob/3c69e22/x/checkers/types/full_game.go#L9-L12]
+    ```go [https://github.com/cosmos/b9-checkers-academy-draft/blob/d7272dd/x/checkers/types/full_game.go#L9-L12]
     func (storedGame *StoredGame) GetCreatorAddress() (creator sdk.AccAddress, err error) {
         creator, errCreator := sdk.AccAddressFromBech32(storedGame.Creator)
         return creator, sdkerrors.Wrapf(errCreator, ErrInvalidCreator.Error(), storedGame.Creator)
     }
     ```
 
-    Plus the same for the [red](https://github.com/cosmos/b9-checkers-academy-draft/blob/3c69e22/x/checkers/types/full_game.go#L14-L17) and [black](https://github.com/cosmos/b9-checkers-academy-draft/blob/3c69e22/x/checkers/types/full_game.go#L19-L22) players.
+    Do the same for the [red](https://github.com/cosmos/b9-checkers-academy-draft/blob/d7272dd/x/checkers/types/full_game.go#L14-L17) and [black](https://github.com/cosmos/b9-checkers-academy-draft/blob/d7272dd/x/checkers/types/full_game.go#L19-L22) players.
 
 2. Parse the game so that it can be played with. The `Turn` has to be set by hand:
```

Comment:
The phrase "play with" is ambiguous:

1. It could mean "adjusted": _We **played with** the system settings_

2. Or it could simply mean "play": _We **played with** the game_ = _We **played** the game_
    (Contrast with: _We **played with** the dog_ =/= _We **played** the dog_ ... no idea what THAT would mean!)

Depending on intent, my recommendations are:

Meaning 1: _Parse the game so that it can be **adjusted**._
Meaning 2: _Parse the game so that it can be **played**._

Reply:
It is meaning 2: so that it can be played on.
Before parsing, we have just a string `"*b*b*b...|..|r*r*..."`. After parsing, we have a proper `Game` object in memory, which has a `Move` function.  Calling this `Move` function is _playing on_ the game. If I got the language right.

Reply:
I would use the simple "played" in this case. Games like chess, checkers, etc, we usually just say _"They **play** a game"_ (also "He **plays** a move" = "She **makes** a move").

Then there are evil English prepositions! In general, some strong collocations are:

play with (a person, usually: "Alice plays with Bob")
play on (a physical surface, "You play chess on a board", or "Football is played on grass") 
play on (a system; "I played it on an XBox")

I'll end "teacher mode" before I get going on phrasal verbs... ;-)